### ISSUE TITLE
EIP bot must error when multiple files are in EIPs asset folder but e.g. only single file is linked in specification document

### ISSUE THREAD

**Issue by TimDaub** [ethereum/EIP-Bot#93]
- Not sure if this is the right repository to file this bug report.
- EIP-4973 is checking in a number of separate files into the /assets/ folder as a reference implementation: https://github.com/ethereum/EIPs/tree/master/assets/eip-4973/src
- However, it seems that when wanting to request those assets as a folder, they're not available: https://eips.ethereum.org/assets/eip-4973/src/
- Only individual files are: https://eips.ethereum.org/assets/eip-4973/src/ERC4973.sol

The EIP bot must check the following:

If an EIP champion checks in multiple files in the asset directory, the EIP bot must check if all of those files are listed in a table in the specification document. Elsewise if e.g. only one file is checked in, the EIP bot must check if this one file is listed in the specification document.

In case e.g. a file is checked into the assets directory but not listed in the specification document, then the EIP bot must fail/throw to indicate to the user that they have to link all documents checked-into the assets folder.

**Comment by MicahZoltu** [comment]
This isn't a bot issue, it has to do with the fact that neither GitHub nor Jekyll markdown renderers (I believe) render folders, they only support linking to files.  For EIPs, the current best option is to either create a manual table of contents markdown file and link to that in your EIP, or just put the table of contents into the EIP itself.

Generally speaking, we try to target the lowest common denominator for markdown rendering so the contents of this repository are maximally compatible with all markdown rendering systems.  In this case, that means no support for folders.

**Comment by TimDaub** [comment]
then IMO tests should be created that check if all files in the assets folder are linked in the EIP. I shipped a folder structure but only linked a single file, only to find later that it lead to bad experiences with people downloading the reference implementation. I fixed it now, but it would have been nice had there been a warning/check.

**Comment by MicahZoltu** [comment]
If you want to update the Title and description of this issue to reflect that suggestion to the bot we can re-open it.  I agree that would be a great addition.

**Comment by TimDaub** [comment]
OK, changed the title and edited the original post.

**Comment by Pandapip1** [comment]
I think this would best be done in the renderer by adding an `index.html` table of contents in every folder and sub-folder of `assets`

**Comment by MicahZoltu** [comment]
> I think this would best be done in the renderer by adding an `index.html` table of contents in every folder and sub-folder of `assets`

We have tried pretty hard in the past to ensure that EIPs repository renders in the lowest common denominator rendering system.  Any tweaks we have made to Jekyll layouts are purely visual, and not functional.  If we allowed people to link to directories and this only worked in the Jekyll rendered site, then any other rendering engine used would break and links would not work properly.
